buildSrc/src/main/kotlin/com/google/devtools/ksp/ApiCheck.kt

```diff
@@ -31,6 +32,16 @@ fun Project.configureMetalava() {
         task.description = "Check API compatibility."
         task.group = "Verification"
         task.args = listOf("--check-compatibility:api:released", API_BASE_FILE) + task.args!!
+        task.inputs.files(API_BASE_FILE).withPropertyName("apiCheckBaseFile").withPathSensitivity(PathSensitivity.RELATIVE)
+
+        val outDir = project.buildDir.resolve("reports/checkApi").takeIf { it.mkdirs() || it.exists() }!!
```

Comment:
Gradle will automatically create output directories if they do not exist.
Manually creating the output directory could disable caching with the message "Gradle does not know how this file was created"

We should use a provider instead to avoid touching the file system during the configuration phase:

`project.layout.buildDirectory.dir("reports/checkApi")